cmd/cluster/logs/cmd.go

```diff
@@ -0,0 +1,145 @@
+package logs
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+const (
+	DynatraceTenantKeyLabel    string = "sre-capabilities.dtp.tenant"
+	HypershiftClusterTypeLabel string = "ext-hypershift.openshift.io/cluster-type"
+	authURL                    string = "https://sso.dynatrace.com/sso/oauth2/token"
+	vaultPath                  string = "osd-sre/dynatrace/sd-sre-platform-oauth-client-grail"
+	vaultAddr                  string = "https://vault.devshift.net"
+)
+
+var (
+	dryRun        bool
+	hcp           bool
+	tail          int
+	hours         int
+	contains      string
+	cluster       string
+	sortOrder     string
+	namespaceList []string
+	nodeList      []string
+	podList       []string
+	containerList []string
+	statusList    []string
+)
+
+func NewCmdLogs() *cobra.Command {
+	logsCmd := &cobra.Command{
+		Use:   "logs",
+		Short: "Fetch logs from Dynatrace",
+		Args:  cobra.NoArgs,
+		RunE:  Run,
+	}
+
+	logsCmd.Flags().BoolVar(&dryRun, "dry-run", false, "Only builds the query without fetching any logs from the tenant")
+	logsCmd.Flags().IntVar(&tail, "tail", 10, "Last 'n' logs to fetch")
+	logsCmd.Flags().IntVar(&hours, "hours", 1, "Number of hours back to search")
+	logsCmd.Flags().StringVar(&contains, "contains", "", "Include logs which contain a phrase")
+	logsCmd.Flags().StringVar(&cluster, "cluster", "", "Cluster name")
+	logsCmd.Flags().StringVar(&sortOrder, "sort", "", "Sort the results by timestamp in either ascending or descending order")
+	logsCmd.Flags().BoolVar(&hcp, "hcp", false, "Set true to Include the HCP Namespace")
+	logsCmd.Flags().StringSliceVar(&namespaceList, "namespace", []string{}, "Namespace(s) (comma-separated)")
+	logsCmd.Flags().StringSliceVar(&nodeList, "node", []string{}, "Node name(s) (comma-separated)")
+	logsCmd.Flags().StringSliceVar(&podList, "pod", []string{}, "Pod name(s) (comma-separated)")
+	logsCmd.Flags().StringSliceVar(&containerList, "container", []string{}, "Container name(s) (comma-separated)")
+	logsCmd.Flags().StringSliceVar(&statusList, "status", []string{}, "Status(Info/Warn/Error) (comma-separated)")
+
+	return logsCmd
+}
+
+func errorExit(err error) {
+	fmt.Println(err)
+	os.Exit(1)
+}
```

Comment:
**suggestion:** As mentioned above, bubble up errors in a go-like manner. We don't need this function, we should always return errors to the main logic. We're using `RunE` so the error won't get lost:

```
	logsCmd := &cobra.Command{
		Use:   "logs",
		Short: "Fetch logs from Dynatrace",
		Args:  cobra.NoArgs,
		RunE:  Run,
	}
```